164 add landing page

maxfukuh4ra opened the pull request:
**_Description:_**
Creating landing page as per Figma design.  Styling is pretty flexible on smaller viewports.

**_New Files:_**
LandingPage.js
LandingPage.module.css

**_Modified File:_** 
AuthWrapper.js now reads Landing.js if the user is not authenticated. From Landing.js the user will be given the choice of "Start Now" or "Sign In". "Sign In" redirects to SignIn as expected. "Start Now" is still an empty button until we make a "Create Account" page or something.

**_Modified Code: (from above)_**
<img width="481" alt="Screenshot 2024-10-29 at 1 13 56 AM" src="https://github.com/user-attachments/assets/68085339-6d43-47fd-bfb7-52df8df2f069">


**_Full-Screen:_**
<img width="1436" alt="Screenshot 2024-10-29 at 12 58 40 AM" src="https://github.com/user-attachments/assets/77e2f108-e645-48da-bfed-ae66cfdcb344">
**_Small-Screen:_**
<img width="706" alt="Screenshot 2024-10-29 at 1 15 53 AM" src="https://github.com/user-attachments/assets/515069ad-f169-47d6-9ec9-a4b23d96dda8">

Fredenck commented on app/components/LandingPage.js:
```diff
@@ -0,0 +1,88 @@
+        <>
+          <div className={styles.titleBar}>
+            <div className={styles.leftTitle}>
+              <h1>BSA | Tennis Consulting</h1>
```
let's try to keep this unified across all pages by updating the Toolbar Component and calling that in layout; I'll make a new issue for this

Fredenck commented on app/styles/LandingPage.module.css:
```diff
@@ -0,0 +1,146 @@
+  margin-top: 1%;
+  font-size: 0.8vw;
+  padding-left: 2%;
+  font-family: 'DM Sans', sans-serif;
```
let's not add custom one-off things here. it's better to define a global font category for header/body

Fredenck commented on app/styles/LandingPage.module.css:
```diff
@@ -0,0 +1,146 @@
+@import url('https://api.fontshare.com/css?f[]=clash-display@100,300,400,500&display=swap');
```
let's not add custom one-off things here. it's better to define a global font category for header/body

Fredenck commented on app/styles/LandingPage.module.css:
```diff
@@ -0,0 +1,146 @@
+
+.mainText {
+  color: #000;
+  font-family: 'Clash Display', sans-serif;
```
same thing here

Fredenck commented on app/styles/LandingPage.module.css:
```diff
@@ -0,0 +1,146 @@
+
+.subText {
+  color: #484848;
+  font-family: 'DM Sans', sans-serif;
```
and here

Fredenck commented on app/styles/LandingPage.module.css:
```diff
@@ -0,0 +1,146 @@
+  transition:
+    background-color 0.3s,
+    transform 0.2s; /* Add smooth transition for shrinking effect */
+  font-family: 'DM Sans', sans-serif;
```
here

Fredenck commented on app/components/LandingPage.js:
```diff
@@ -0,0 +1,88 @@
+            </div>
+            <div className={styles.rightTitle}>
+              {authUser ? (
+                <button onClick={handleSignOut}>Sign Out</button>
```
this is never used because if we are signed in, we are routed to the Dashboard component. Don't worry about handling this, Rushil has the correct changes in his PR

maxfukuh4ra replied:
wait yeah sorry i just kept what it had previously even tho it wasn't really doing anything lol

Fredenck reviewed (changes requested):
overall great work, but make sure every line you write has a purpose. The sign out feature is not able to be used because if a user is signed in, they are routed to the Dashboard page. Furthermore, it doesn't make sense to have a sign out button on the landing page; it should just have the sign-in functionality. 

please take out unecessary ad-hoc CSS. 

Logic is great though :)

maxfukuh4ra commented:
Removed the sign out functionality and also moved font-family styling over to global.css, sorry I didn't catch that beforehand.
In the future, will adjust global.css accordingly as needed for styling other pages but should make sense for now.

Fredenck reviewed (approved):
lgtm
